ultralytics/utils/ops.py

```diff
@@ -301,35 +304,35 @@ def clip_coords(coords, shape):
         coords[..., 1] = coords[..., 1].clip(0, shape[0])  # y
 
 
-def scale_image(masks, im0_shape, ratio_pad=None):
+def scale_image(masks, img0_shape, ratio_pad=None):
     """
     Takes a mask, and resizes it to the original image size
 
     Args:
         masks (np.ndarray): resized and padded masks/images, [h, w, num]/[h, w, 3].
-        im0_shape (tuple): the original image shape
+        img0_shape (tuple): the original image shape
         ratio_pad (tuple): the ratio of the padding to the original image.
 
     Returns:
         masks (torch.Tensor): The masks that are being returned.
     """
     # Rescale coordinates (xyxy) from im1_shape to im0_shape
-    im1_shape = masks.shape
-    if im1_shape[:2] == im0_shape[:2]:
+    img1_shape = masks.shape
+    if img1_shape[:2] == img0_shape[:2]:
         return masks
     if ratio_pad is None:  # calculate from im0_shape
-        gain = min(im1_shape[0] / im0_shape[0], im1_shape[1] / im0_shape[1])  # gain  = old / new
-        pad = (im1_shape[1] - im0_shape[1] * gain) / 2, (im1_shape[0] - im0_shape[0] * gain) / 2  # wh padding
+        gain = min(img1_shape[1] / img0_shape[1], img1_shape[0] / img0_shape[0])  # gain = old / new
+        pad = (img1_shape[1] - math.ceil(img0_shape[1] * gain)) // 2, (
+            img1_shape[0] - math.ceil(img0_shape[0] * gain)) // 2  # wh padding
```

Comment:
@Wovchena should we compute `gain` and `pad` here the same way as in ops.py L734-L737? This computation differs from the other two, which are identical.